### PR TITLE
Display sound as an alert when a Pokemon appears

### PR DESCRIPTION
**Pull request by NgJason**
## Description
1) -s Play sound to alert the user via console when a desire Pokemon is discovered.
2) -sw Play sound to alert the user via browser when a desire Pokemon is discovered.

Usage:
pip install --upgrade -r requirements.tx
To activate audio for console, add -s flag.
To activate audio for console, add -sw flag.

## Motivation and Context
This is especially useful for users who multitask and have this program run in the background.
https://github.com/AHAAAAAAA/PokemonGo-Map/pull/470

## How Has This Been Tested?
Executed program locally.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by mkody** [comment]
I'm not sure if playing the audio from the script would be a great idea. I'd prefer to use it from the browser. (You know, you can just open a tab and leave it pinned in the background.)

**Comment by rezz3x** [comment]
I would like if smth like a push notifications would come if a uncommon/rare Pokemon is on the map.

**Comment by SSminowjohnson** [comment]
This is exactly what I was looking for. I have -o set to only the Pokemon I still need to discover, and added "<meta http-equiv="refresh" content="10">" to auto refresh the map every 10 seconds, but unless I'm staring at the map I may still miss rare Pokemon. Is there any way for me to acquire this feature at this time?

File: pogom/utils.py [hunk 1]
```diff
@@ -43,6 +43,9 @@ def get_args():
     parser.add_argument('-d', '--debug', help='Debug Mode', action='store_true')
     parser.add_argument('-m', '--mock', help='Mock mode. Starts the web server but not the background thread.', action='store_true', default=False)
     parser.add_argument('-k', '--google-maps-key', help='Google Maps Javascript API Key', default=None, dest='gmaps_key')
+    groupA = parser.add_mutually_exclusive_group(required=False)
+    groupA.add_argument('-s','--play-sound',help='Play an alert in the console when a Pokemon is visible',action='store_true',default=False)
+    #groupA.add_argument('-sw','--play-sound-web',help='Play an alert in the browser when a Pokemon is visible',action='store_true',default=False)
```
Comment (calebthompson) [inline pogom/utils.py]: Can this be deleted?

**Comment by NgJason** [comment]
Hi @mkody @rezz3x @SSminowjohnson , I am waiting for -ar (auto refresh) to complete before I can implement any browser side notifications. 

I have actually already implemented this to the master (production) before the rewrite, it displays sound notification via the browser. If you are using the production (master branch), you can grab the code from https://github.com/AHAAAAAAA/PokemonGo-Map/pull/470.

Support with :+1: ? Thanks.

**Comment by mkody** [comment]
Thanks for the response. And yes, giving users more choices isn't bad anyways. :grimacing:

File: pogom/utils.py (hunk 1 above)
Comment (NgJason) [inline pogom/utils.py]: @calebthompson  I am new pretty new to Python myself, but I don't believe so. The idea is to have -s xor -sw (will be implemented when -ar is completed). (exclusive or, not both -s and -sw can be on at the same time)

Comment (calebthompson) [inline pogom/utils.py]: This line is commented out, so it's not executed. What I mean is, instead of commenting it out can you delete it?

Comment (NgJason) [inline pogom/utils.py]: @calebthompson I am leaving it there for now to indicate that I plan to implement -sw before this gets merged. Waiting on -ar before I can complete -sw though. But you are right, at this current time, it is useless.

File: pogom/models.py [hunk 2]
```diff
@@ -87,6 +83,15 @@ def parse_map(map_dict):
...
         for p in cell.get('wild_pokemons', []):
+            pokemon_name = get_pokemon_name(p['pokemon_data']['pokemon_id'])
+            pokemon_id = str(p['pokemon_data']['pokemon_id'])
+            if args.ignore:
+                if pokemon_name in args.ignore or pokemon_id in args.ignore:
+                    continue
+            if args.only:
+                if pokemon_name not in args.only and pokemon_id not in args.only:
+                    continue
+            play()
```
Comment (MonkeyLeeT) [inline pogom/models.py]: Why do you wanna move ignore/only checking logics here? Is there any reason and did you test with ignore/only lists of Pokemon names/ids (also mixed if possible?)

Comment (NgJason) [inline pogom/models.py]: @MonkeyLeeT I've only tested with name/ids in the --only list. Servers seem to be down for me right now. I have pulled this code out in to it's own pull request. It has some performance improvement and this will allow the console to play sound. More information: https://github.com/AHAAAAAAA/PokemonGo-Map/pull/658

**Comment by SSminowjohnson** [comment]
Sorry to be a pain but not sure what I'm doing wrong here. I downloaded the 6 changed files, placed them all in the root folder (with models.py, sound.py, utils.py, in a newly created "pogom" folder and map.html in the "templates" folder), ran pip install --upgrade -r requirements.txt which seems to have updated correctly, but every time I add -s anywhere I get the following error:

example.py: error: argument -st/--step-limit: expected one argument

**Comment by NgJason** [comment]
@SSminowjohnson , if you are following the wiki, you should check out this fork instead: #470. The wiki is still set to the Master branch. The updates in this fork is for the development branch. 

In other words: 
Master branch: https://github.com/AHAAAAAAA/PokemonGo-Map/archive/master.zip + #470
Development branch: https://github.com/AHAAAAAAA/PokemonGo-Map/archive/develop.zip + this fork. Replacing the files should be sufficient.

**Comment by SSminowjohnson** [comment]
Ah I'm a damn fool. Thanks so much

**Comment by NgJason** [comment]
@mkody  @rezz3x  @SSminowjohnson Play a sound alert via web is implemented. I can only run limited testing due to an unrelated open issue https://github.com/AHAAAAAAA/PokemonGo-Map/issues/699. 

Any feedback is welcomed! Cheers.

**Comment by AHAAAAAAA** [comment]
I had to do `brew install portaudio` for this to work alongside the pip file, seems to introduce unnecessary complexity. Is there a way to instead do this in the web browser instead? Also -sw doesn't work.

**Comment by NgJason** [comment]
@AHAAAAAAA I just fixed a bug in my code that involved the -o and -i list, this might be why it didn't work for you. Can you let me know if this is working for you in the browser now? 

I am using Chrome V 51.0.2704.103 m.  Example:
 python runserver.py -a google -u XXXXXXXXXXX@gmail.com -p XXXXXX -o "Zubat" -l "37.788573,-122.396042" -st 10 -sw

File: pogom/utils.py [hunk 3]
```diff
@@ -43,6 +43,9 @@ def get_args():
     parser.add_argument('-d', '--debug', help='Debug Mode', action='store_true')
     parser.add_argument('-m', '--mock', help='Mock mode. Starts the web server but not the background thread.', action='store_true', default=False)
     parser.add_argument('-k', '--google-maps-key', help='Google Maps Javascript API Key', default=None, dest='gmaps_key')
+    groupA = parser.add_mutually_exclusive_group(required=False)
+    groupA.add_argument('-s','--play-sound',help='Play an alert in the console when a Pokemon is visible',action='store_true',default=False)
+    groupA.add_argument('-sw','--play-sound-web',help='Play an alert in the browser when a Pokemon is visible',action='store_true',default=False)
```
Comment (Tigershark2005) [inline pogom/utils.py]: --PLAY_WEB_SOUND

Comment (NgJason) [inline pogom/utils.py]: @Tigershark2005 there were actually 2 line in runserver,py:
if args.play_sound_web:
        config['PLAY_WEB_SOUND'] = 1

but I have updated it so it looks better for uniformity.

**Comment by Tigershark2005** [comment]
Right on. This would be a great merge with the pull request for notification for a certain pokemon. That thread includes check boxes and such. This would fit in well as far as the sound module and would just need to trigger based off their checkboxes and probably an additional that would allow your sound to play.

**Comment by NgJason** [comment]
@Tigershark2005 Things are a little weird at the moment because -i -o was removed some commits ago.
